ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedWindowTableIQv2Test.java

```diff
@@ -0,0 +1,321 @@
+
+  @Before
+  public void setUp() {
+    table = new KsMaterializedWindowTableIQv2(stateStore, WINDOW_SIZE);
+
+    when(stateStore.getStateStoreName()).thenReturn("rocksdb-store");
+    when(stateStore.getKafkaStreams()).thenReturn(kafkaStreams);
+    when(stateStore.schema()).thenReturn(SCHEMA);
+  }
+
+  @SuppressWarnings("UnstableApiUsage")
+  @Test
+  public void shouldThrowNPEs() {
+    new NullPointerTester()
+        .setDefault(KsStateStore.class, stateStore)
+        .testConstructors(KsMaterializedWindowTable.class, Visibility.PACKAGE);
+  }
+
+  @Test
+  public void shouldThrowIfGettingStateStoreFails() {
+    // Given:
+    when(kafkaStreams.query(any())).thenThrow(new MaterializationTimeOutException("Boom"));
+
+    // When:
+    final Exception e = assertThrows(
+        MaterializationException.class,
+        () -> table.get(A_KEY, PARTITION, WINDOW_START_BOUNDS, WINDOW_END_BOUNDS)
+    );
+
+    // Then:
+    assertThat(e.getMessage(), containsString(
+        "Failed to get value from materialized table"));
+    assertThat(e.getCause(), (instanceOf(MaterializationTimeOutException.class)));
+  }
+
+  @Test
+  public void shouldThrowIfGettingStateStoreFailedResult() {
+    // Given:
+    final StateQueryResult partitionResult = new StateQueryResult();
+    partitionResult.addResult(PARTITION, QueryResult.forFailure(FailureReason.STORE_EXCEPTION, "Boom"));
+    when(kafkaStreams.query(any())).thenReturn(partitionResult);
+
+    // When:
+    final Exception e = assertThrows(
+      MaterializationException.class,
+      () -> table.get(A_KEY, PARTITION, WINDOW_START_BOUNDS, WINDOW_END_BOUNDS)
+    );
+
+    // Then:
+    assertThat(e.getMessage(), containsString(
+      "Failed to get value from materialized table"));
+    assertThat(e.getCause(), (instanceOf(MaterializationException.class)));
+  }
+
+  @Test
+  public void shouldThrowIfGettingStateStoreFails_fetchAll() {
+    // Given:
+    when(stateStore.getKafkaStreams()).thenThrow(new MaterializationTimeOutException("Boom"));
+
+    // When:
+    final Exception e = assertThrows(
+        MaterializationException.class,
+        () -> table.get(PARTITION, WINDOW_START_BOUNDS, WINDOW_END_BOUNDS)
+    );
+
+    // Then:
+    assertThat(e.getMessage(), containsString(
+        "Failed to scan materialized table"));
+    assertThat(e.getCause(), (instanceOf(MaterializationTimeOutException.class)));
+  }
+
+  @Test
+  public void shouldThrowIfStoreFetchFails() {
+    // Given:
+    when(kafkaStreams.query(any()))
+        .thenThrow(new MaterializationTimeOutException("Boom"));
+
+    // When:
+    final Exception e = assertThrows(
+        MaterializationException.class,
+        () -> table.get(A_KEY, PARTITION, WINDOW_START_BOUNDS, WINDOW_END_BOUNDS)
+    );
+
+    // Then:
+    assertThat(e.getMessage(), containsString(
+        "Failed to get value from materialized table"));
+    assertThat(e.getCause(), (instanceOf(MaterializationTimeOutException.class)));
+  }
```

Comment:
This test is a duplicate of `shouldThrowIfGettingStateStoreFails` above, although in retrospect, I guess that other test is a duplicate of this one, since the name if this test is more accurate.

Reply:
Good catch! Adjusted the test